pretyping/typeclasses.ml

```diff
@@ -263,12 +263,14 @@ let resolve_typeclasses ?(filter=no_goals) ?(unique=get_typeclasses_unique_solut
 let error_unresolvable env evd comp =
   let exception MultipleFound in
   let fold ev accu =
-    let evi = Evd.find_undefined evd ev in
-    let ev_class = class_of_constr env evd (Evd.evar_concl evi) in
-    if Option.is_empty ev_class then accu
-    else (* focus on one instance if only one was searched for *)
-    if Option.has_some accu then raise MultipleFound
-    else (Some ev)
+    match Evd.find_undefined evd ev with
+    | exception Not_found -> None
+    | evi ->
+      let ev_class = class_of_constr env evd (Evd.evar_concl evi) in
+      if Option.is_empty ev_class then accu
+      else (* focus on one instance if only one was searched for *)
+      if Option.has_some accu then raise MultipleFound
+      else (Some ev)
```

Comment:
why this change?

Reply:
This is a bugfix

Reply:
It would otherwise result in an uncaught exception

Reply:
Why though? What's the test case?

Reply:
I guess the evar in question could be solved by a previous TC resolution. Does this make sense?

Reply:
Yes, it could be solved from another resolution in principle.